### PR TITLE
feature/loginlogout

### PR DESCRIPTION
**Pull request by jaeseong-kim**
### 변경사항

**AS-IS**
 회원가입 기능
**TO-BE**
 로그인, 로그아웃 기능 추가

### CODE REVIEW

**Comment by jaeseong-kim** [comment]
|File|Coverage [38.97%]|:x:|
|:-|:-:|:-:|
|[JasyptConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/4f393830c65dafa0c1cb0c1edcce65ce11207154/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FJasyptConfiguration.java)|100%|:green_apple:|
|[SecurityConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/4f393830c65dafa0c1cb0c1edcce65ce11207154/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FSecurityConfiguration.java)|100%|:green_apple:|
|[UserController.java](https://github.com/jaeseong-kim/sharingcar/blob/4f393830c65dafa0c1cb0c1edcce65ce11207154/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fcontroller%2FUserController.java)|16.67%|:x:|
|[UserAuthenticationFailureHandler.java](https://github.com/jaeseong-kim/sharingcar/blob/4f393830c65dafa0c1cb0c1edcce65ce11207154/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FUserAuthenticationFailureHandler.java)|11.54%|:x:|
|[UserServiceImpl.java](https://github.com/jaeseong-kim/sharingcar/blob/4f393830c65dafa0c1cb0c1edcce65ce11207154/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fservice%2Fimpl%2FUserServiceImpl.java)|5.81%|:x:|
|[EmailNotAuthException.java](https://github.com/jaeseong-kim/sharingcar/blob/4f393830c65dafa0c1cb0c1edcce65ce11207154/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fexception%2FEmailNotAuthException.java)|0%|:x:|

|Total Project Coverage|9.72%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/example/sharingcar/user/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -66,4 +74,23 @@ public boolean authorizeEmail(String uuid) {
 
 		return true;
 	}
+
+	@Override
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+		Optional<User> optionalUser = userRepository.findById(username);
```
Comment (KimHyoJin) [inline src/main/java/com/example/sharingcar/user/service/impl/UserServiceImpl.java]: userRepository.findById(username).orElseThrow(() -> throw new UsernameNotFoundException("회원 정보가 존재하지 않습니다."));
로 대체할수 있을 것 같습니다

File: src/main/java/com/example/sharingcar/configuration/UserAuthenticationFailureHandler.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler;
+
+public class UserAuthenticationFailureHandler extends SimpleUrlAuthenticationFailureHandler {
+
+	@Override
+	public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response,
+		AuthenticationException exception) throws IOException, ServletException {
+
+		String msg = "아이디 및 비밀번호가 일치하지 않습니다.";
```
Comment (KimHyoJin) [inline src/main/java/com/example/sharingcar/configuration/UserAuthenticationFailureHandler.java]: string보단 enum형태로 관리하면 좀더 코드가 깔끔할 것 같습니다

**Review by KimHyoJin — APPROVED** [review]
안녕하세요 밍밍멘토입니다. 몇가지 코멘트 드렸어요! 수고많으셨습니다

**Comment by jaeseong-kim** [comment]
|File|Coverage [12.31%]|:x:|
|:-|:-:|:-:|
|[JasyptConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FJasyptConfiguration.java)|100%|:green_apple:|
|[SecurityConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FSecurityConfiguration.java)|100%|:green_apple:|
|[UserController.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fcontroller%2FUserController.java)|16.67%|:x:|
|[UserAuthenticationFailureHandler.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FUserAuthenticationFailureHandler.java)|11.54%|:x:|
|[UserServiceImpl.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fservice%2Fimpl%2FUserServiceImpl.java)|5.26%|:x:|
|[User.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fentity%2FUser.java)|0%|:x:|
|[EmailNotAuthException.java](https://github.com/jaeseong-kim/sharingcar/blob/db34d9478c2fc696e411582f879be05c700ac652/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fexception%2FEmailNotAuthException.java)|0%|:x:|

|Total Project Coverage|9.36%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/example/sharingcar/configuration/UserAuthenticationFailureHandler.java (hunk 2 above)
Comment (jaeseong-kim) [reply src/main/java/com/example/sharingcar/configuration/UserAuthenticationFailureHandler.java]: 수정했습니다! 피드백 감사합니다!

**Comment by jaeseong-kim** [comment]
|File|Coverage [13.38%]|:x:|
|:-|:-:|:-:|
|[JasyptConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FJasyptConfiguration.java)|100%|:green_apple:|
|[SecurityConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FSecurityConfiguration.java)|100%|:green_apple:|
|[UserController.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fcontroller%2FUserController.java)|16.67%|:x:|
|[UserAuthenticationFailureHandler.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FUserAuthenticationFailureHandler.java)|11.54%|:x:|
|[UserServiceImpl.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fservice%2Fimpl%2FUserServiceImpl.java)|5.39%|:x:|
|[User.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fentity%2FUser.java)|0%|:x:|
|[ErrorType.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Ferror%2FErrorType.java)|0%|:x:|
|[EmailNotAuthException.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fexception%2FEmailNotAuthException.java)|0%|:x:|

|Total Project Coverage|10.22%|:x:|
|:-|:-:|:-:|

**Comment by jaeseong-kim** [comment]
|File|Coverage [13.38%]|:x:|
|:-|:-:|:-:|
|[JasyptConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FJasyptConfiguration.java)|100%|:green_apple:|
|[SecurityConfiguration.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FSecurityConfiguration.java)|100%|:green_apple:|
|[UserController.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fcontroller%2FUserController.java)|16.67%|:x:|
|[UserAuthenticationFailureHandler.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fconfiguration%2FUserAuthenticationFailureHandler.java)|11.54%|:x:|
|[UserServiceImpl.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fservice%2Fimpl%2FUserServiceImpl.java)|5.39%|:x:|
|[User.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fentity%2FUser.java)|0%|:x:|
|[ErrorType.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Ferror%2FErrorType.java)|0%|:x:|
|[EmailNotAuthException.java](https://github.com/jaeseong-kim/sharingcar/blob/8b9bf33db51e1433a3f7c8912e81142dcebe1122/src%2Fmain%2Fjava%2Fcom%2Fexample%2Fsharingcar%2Fuser%2Fexception%2FEmailNotAuthException.java)|0%|:x:|

|Total Project Coverage|10.22%|:x:|
|:-|:-:|:-:|

**Pull request merged**